libavcodec/qsvenc.c

```diff
@@ -494,10 +500,10 @@ static int init_video_param(AVCodecContext *avctx, QSVEncContext *q)
     q->param.mfx.FrameInfo.BitDepthChroma = desc->comp[0].depth;
     q->param.mfx.FrameInfo.Shift          = desc->comp[0].depth > 8;
 
-    // TODO:  detect version of MFX--if the minor version is greater than
-    // or equal to 19, then can use the same alignment settings as H.264
-    // for HEVC
-    q->width_align = avctx->codec_id == AV_CODEC_ID_HEVC ? 32 : 16;
+    // If the minor version is greater than or equal to 19,
+    // then can use the same alignment settings as H.264 for HEVC
+    q->width_align = (avctx->codec_id != AV_CODEC_ID_HEVC ||
```

Comment:
Where this requirement was taken from? Mediasdk requirement is: "Width must be a multiple of 16. Height must be a multiple of 16 for progressive frame sequence and a multiple of 32 otherwise."

To me what is programmed here looks completely incorrect.

Reply:
I guess some old version of MSDK weren't totally  following the documentation you mentioned. 
width_alignment of hevc is 32 from the beginning of Luca's enabling: https://code.oxygene.sk/mirrors/ffmpeg/commit/66acb76bb0492b263215ca9b4d927a7be39ace02#b8b2636776a36ebc6cd1e21f1ec8d7dfac28dc6f_52_52 
And Aaron added such a comment in  ae5b67ee64bec00c52881ff8426fddaf477341c1 
To confirm this, we may need to check some old version MSDK. 
But anyway, for current MSDK, even for hevc encoding, 16 bytes is enough and this is the patch is doing. (I admit that I am conservative that don't change to 16 bytes totally to keep compatibility, because it is not easy to check old version MSDK for me and I chose to believe Luca and Aaron them definitely found 16 bytes aligned was not enough for old version MSDK).
BTW,  comment is always welcome, but should be better discuss in ffmpeg-devel.